src/NServiceBus.Msmq.AcceptanceTests/When_distributing_an_event.cs

```diff
@@ -0,0 +1,120 @@
+﻿namespace NServiceBus.AcceptanceTests
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using AcceptanceTesting.Customization;
+    using AcceptanceTesting;
+    using Features;
+    using NServiceBus.Routing;
+    using NUnit.Framework;
+    using Support;
+
+    public class When_distributing_an_event : NServiceBusAcceptanceTest
+    {
+        static string PublisherEndpoint => Conventions.EndpointNamingConvention(typeof(Publisher));
+        static string SubscriberEndpoint => Conventions.EndpointNamingConvention(typeof(Subscriber));
+
+        [Test]
+        public async Task Should_round_robin()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<Publisher>(b => b.When(c => c.Subscriptions == 2, async (session, c) =>
+                {
+                    for (var i = 0; i < 10; i++)
+                    {
+                        await session.Publish(new MyEvent()).ConfigureAwait(false);
+                    }
+                }))
+                .WithEndpoint<Subscriber1>()
+                .WithEndpoint<Subscriber2>()
+                .Done(c => c.Receiver1TimesCalled + c.Receiver2TimesCalled>= 10)
+                .Run(TimeSpan.FromSeconds(20));
+
+            Assert.That(context.Receiver1TimesCalled + context.Receiver2TimesCalled, Is.EqualTo(10));
+        }
+
+        public class Context : ScenarioContext
+        {
+            public int Subscriptions { get; set; }
+            public int Receiver1TimesCalled => receiver1TimesCalled;
+            public int Receiver2TimesCalled => receiver2TimesCalled;
+
+            public int receiver1TimesCalled;
+            public int receiver2TimesCalled;
+        }
+
+        public class Publisher : EndpointConfigurationBuilder
+        {
+            public Publisher()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.DisableFeature<AutoSubscribe>();
+                    c.OnEndpointSubscribed<Context>((s, context) =>
+                    {
+                        context.Subscriptions++;
+                    });
+                    c.LimitMessageProcessingConcurrencyTo(1);
+                });
+            }
+        }
+
+        //Used only to generate a name
+        public class Subscriber : EndpointConfigurationBuilder
+        {
+        }
+
+        public class Subscriber1 : EndpointConfigurationBuilder
+        {
+            public Subscriber1()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    var transport = c.UseTransport<MsmqTransport>();
+                    transport.RegisterPublisherForType(PublisherEndpoint, typeof(MyEvent));
+                    transport.AddAddressTranslationException(new EndpointInstance(SubscriberEndpoint).AtMachine(RuntimeEnvironment.MachineName), SubscriberEndpoint + "-1");
```

Comment:
I'm assuming you've used `AddAddressTranslationException` here instead of just setting an instance discriminator because then this forces the subscriptions to use an instance queue and not a shared queue?